crypto-core/src/main/java/com/palantir/crypto2/io/ApacheCtrDecryptingSeekableInput.java

```diff
@@ -110,8 +112,38 @@ public void seek(long position) throws IOException {
 
         @Override
         public void close() throws IOException {
+            readBuffer = null;
             input.close();
         }
-    }
 
+        private void resize(int required) {
+            int size = readBuffer.length;
+            readBuffer = new byte[newLength(size, required - size, size << 1)];
+        }
+
+        // copied from jdk.internal.util.ArraysSupport
```

Comment:
In't that GPL?

Reply:
bleh, yes it is. I'll rewrite & simplify a bit